### PR TITLE
Inaccessible project display

### LINKED ISSUE #2366: Polish inaccessible project display
From https://github.com/toggl/mobileapp/pull/2351

- is it OK to show the "(inaccessible project)" until we fetch the project?
- I reused the "no project" color (dark gray) - is that OK?

More details later/upon request.

### PR DESCRIPTION
**Pull request by juanlaube**
Closes #2366 

🦑 whenever

### CODE REVIEW

File: Toggl.Foundation/Extensions/ThreadSafeExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        public static string DisplayName(this IThreadSafeProject project)
+        {
+            var name = project.Name ?? "";
+
+            if (name == Resources.InaccessibleProject)
+            {
+                return name;
+            }
+
+            return project.Active ? name : $"{name} (archived)";
```
Comment (sofokli1) [inline Toggl.Foundation/Extensions/ThreadSafeExtensions.cs]: Why not put `(archived)` in Resources?

File: Toggl.Foundation/Extensions/ThreadSafeExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    public static class ThreadSafeExtensions
+    {
+        private const string archivedColor = "#cecece";
+
+        public static string DisplayName(this IThreadSafeProject project)
+        {
+            var name = project.Name ?? "";
+
+            if (name == Resources.InaccessibleProject)
```
Comment (amulware) [inline Toggl.Foundation/Extensions/ThreadSafeExtensions.cs]: I'm not liking this check, here and below. Is there no safer way to do this based on the project syncstate maybe? In fact, I'd say that if the project is a ghost, we shouldn't even consider it's name and just return the resource string maybe?

Comment (juanlaube) [reply Toggl.Foundation/Extensions/ThreadSafeExtensions.cs]: I don't like this either and was expecting this to appear here :) I will see if there's anything else that gives us a hint of ghost projects, but so far I couldn't find anything

Comment (amulware) [reply Toggl.Foundation/Extensions/ThreadSafeExtensions.cs]: Check the PRs on the subject, they use a different sync state if I'm not mistaken.

File: Toggl.Foundation/Extensions/ThreadSafeExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
+﻿using Toggl.Foundation.Models.Interfaces;
+
+namespace Toggl.Foundation.Extensions
+{
+    public static class ThreadSafeExtensions
+    {
+        private const string archivedColor = "#cecece";
```
Comment (amulware) [inline Toggl.Foundation/Extensions/ThreadSafeExtensions.cs]: We don't actually use this colour for archived projects, so I think it's better to name the constant `inaccessibleProjectColor` or similar.

File: Toggl.Foundation/Extensions/ThreadSafeExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+        public static string DisplayName(this IThreadSafeProject project)
+        {
+            var name = project.Name ?? "";
+
+            switch (project.SyncStatus)
+            {
+                case PrimeRadiant.SyncStatus.RefetchingNeeded:
+                    return name;
```
Comment (amulware) [inline Toggl.Foundation/Extensions/ThreadSafeExtensions.cs]: I think we should return a constant string from resources here and not store the name in the database (we can maybe remove it from there later). Unless there is some point in having it in the database, but I don't see any.

**Pull request merged**